vis/graphs.js

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable no-undef */
+
+var num_projects = [];
+var y_descriptions = [];
+var type_quarter = [];
+
+const number_project_types = 10;
+const number_quarters = 12;
+
+d3.json('projects-extended.json').then((res) => {
+    Object.keys(res).forEach(year => {
+        const quarters = Object.keys(res[year]).reverse();
+        const quartersinfo = Object.values(res[year]).reverse();
+        for(let i = 0; i < Object.values(quarters).length; i++){
+            let quarter = year + ' ' + Object.values(quarters)[i];
+            type_quarter.push(quarter);
+            num_projects.push(Object.keys(Object.values(quartersinfo)[i]).length-3);
+
+            let project = Object.values(quartersinfo)[i];
+            project_desc = Object.values(project).filter(x => x.Description);
+            y_descriptions.push(project_desc);
+        }
+    });
+
+    let proj_type_num = Array(number_project_types).fill(0);
+
+    let proj_type_quarter = Array(number_quarters).fill().map(() =>
+        Array(number_project_types).fill(0),
+    );
+
+    let quarter_count = 0;
+    y_descriptions.map(quarter => {
+        quarter.map(desc => {
+            for (i in desc) {
+                if(desc[i].includes('tech')) {
+                    proj_type_num[0]++;
+                    proj_type_quarter[quarter_count][0]++;
+                }
+                else if (desc[i].includes('non')){
+                    proj_type_num[1]++;
+                    proj_type_quarter[quarter_count][1]++;
+                }
+                if(desc[i].includes('web')) {
+                    proj_type_num[7]++;
+                    proj_type_quarter[quarter_count][7]++;
+                }
+                if(desc[i].includes('app')) {
+                    proj_type_num[4]++;
+                    proj_type_quarter[quarter_count][4]++;
+                }
+                if(desc[i].includes('art')) {
+                    proj_type_num[3]++;
+                    proj_type_quarter[quarter_count][3]++;
+                }
+                else if(desc[i].includes('ar')) {
+                    proj_type_num[8]++;
+                    proj_type_quarter[quarter_count][8]++;
+                }
+                if(desc[i].includes('game')) {
+                    proj_type_num[5]++;
+                    proj_type_quarter[quarter_count][5]++;
+                }
+                if(desc[i].includes('writing')) {
+                    proj_type_num[2]++;
+                    proj_type_quarter[quarter_count][2]++;
+                }
+                if(desc[i].includes('iot')) {
+                    proj_type_num[6]++;
+                    proj_type_quarter[quarter_count][6]++;
+                }
+                if(desc[i].includes('vr')) {
+                    proj_type_num[9]++;
+                    proj_type_quarter[quarter_count][9]++;
+                }
+            }
+        });
+        quarter_count++;
+    });
+
+    let data_projects = [];
+    let index = 2;
```

Comment:
remove this index here and add it to the map function as stated in the following comment